components/org.wso2.carbon.identity.api.server.script.library/org.wso2.carbon.identity.api.server.script.library.v1/src/main/java/org/wso2/carbon/identity/api/server/script/library/v1/core/ServerScriptLibrariesService.java

```diff
@@ -0,0 +1,439 @@
+        }
+        if (isScriptLibraryAvailable(scriptLibraryPOSTRequest.getName())) {
+            throw handleScriptLibraryClientError(Constants.ErrorMessage.ERROR_SCRIPT_LIBRARY_ALREADY_FOUND,
+                    Response.Status.CONFLICT, scriptLibraryPOSTRequest.getName(),
+                    ContextLoader.getTenantDomainFromContext());
+        } else {
+            FunctionLibrary functionLibrary = createScriptLibrary(scriptLibraryPOSTRequest);
+            try {
+                if (scriptLibraryPOSTRequest.getName().contains(Constants.SCRIPT_LIBRARY_EXTENSION)) {
+                    ScriptLibraryServiceHolder.getScriptLibraryManagementService()
+                            .createFunctionLibrary(functionLibrary,
+                                    ContextLoader.getTenantDomainFromContext());
+                } else {
+                    throw handleScriptLibraryClientError(Constants.ErrorMessage.ERROR_SCRIPT_LIBRARY_NAME_VALIDATION,
+                            Response.Status.BAD_REQUEST);
+                }
+            } catch (FunctionLibraryManagementException e) {
+                throw handleScriptLibraryError(e, Constants.ErrorMessage.ERROR_CODE_ERROR_ADDING_SCRIPT_LIBRARY);
+            }
+        }
+
+    }
+
+    /**
+     * Update a script library identified by resource ID.
+     *
+     * @param scriptLibraryName  Name of the script library.
+     * @param contentInputStream Content of the script library code.
+     * @param description        Description of the script library
+     * @return ScriptLibrary.
+     */
+    public ScriptLibraryResponse updateScriptLibrary(String scriptLibraryName, InputStream contentInputStream,
+                                                     String description) {
+
+        ScriptLibraryPUTRequest scriptLibraryPUTRequest = new ScriptLibraryPUTRequest();
+        scriptLibraryPUTRequest.setDescription(description);
+        try {
+            scriptLibraryPUTRequest.setContent(
+                    new File(IOUtils.toString(contentInputStream, StandardCharsets.UTF_8.name())));
+        } catch (IOException e) {
+            log.error("Error occurred while reading contentInputStream: " + e);
+            throw handleScriptLibraryClientError(Constants.ErrorMessage.ERROR_CODE_ERROR_UPDATING_SCRIPT_LIBRARY,
+                    Response.Status.INTERNAL_SERVER_ERROR);
+        }
+        if (isScriptLibraryAvailable(scriptLibraryName)) {
+            FunctionLibrary functionLibrary = createScriptLibraryPut(scriptLibraryName, scriptLibraryPUTRequest);
+            try {
+                ScriptLibraryServiceHolder.getScriptLibraryManagementService()
+                        .updateFunctionLibrary(scriptLibraryName, functionLibrary,
+                                ContextLoader.getTenantDomainFromContext());
+            } catch (FunctionLibraryManagementException e) {
+                throw handleScriptLibraryError(e, Constants.ErrorMessage.ERROR_CODE_ERROR_UPDATING_SCRIPT_LIBRARY
```

Comment:
can't we inline 207 and 208

Reply:
[Fixed commit](https://github.com/wso2/identity-api-server/pull/81/commits/f1f24a495063022b48d0fe2a91be8b83c6cf07de)